core/src/main/scala/org/elasticmq/actor/queue/QueueActorWaitForMessagesOps.scala

```diff
@@ -67,13 +67,15 @@ trait QueueActorWaitForMessagesOps extends ReplyingActor with QueueActorMessageO
            AwaitingData(originalSender, ReceiveMessages(visibilityTimeout, count, _, receiveRequestAttemptId), _))) =>
         val received = super.receiveMessages(visibilityTimeout, count, receiveRequestAttemptId)
 
-        if (received != Nil) {
-          originalSender ! received
-          logger.debug(
-            s"${queueData.name}: Awaiting messages: replying to sequence $seq with ${received.size} messages.")
-          awaitingReply.remove(seq)
+        received match {
+          case Right(receivedList) if receivedList.nonEmpty =>
+            originalSender ! received
+            logger.debug(
+              s"${queueData.name}: Awaiting messages: replying to sequence $seq with ${receivedList.size} messages.")
+            awaitingReply.remove(seq)
 
-          tryReply()
+            tryReply()
+          case _ => ()
```

Comment:
and what if there are requests waiting for receiving a message, but the limit is hit? Does deleting a message trigger responding to waiting requests as well?